### PR TITLE
Add new annotations table and ORM model

### PR DESCRIPTION
**Pull request by jfcalvo**
# Description

This PR adds the following changes:
* A new migration to create `annotations` table. This is a temporal name if we think a better one is possible we can change it later.
* A new `Annotation` ORM model with the required relationships.
* Some missing details from previous PRs.

### CODE REVIEW

File: src/argilla/server/models.py [hunk 1]
```diff
@@ -34,26 +34,54 @@ def default_inserted_at(context):
...
     annotator = "annotator"
 
 
+class Annotation(Base):
+    __tablename__ = "annotations"
+
+    id: Mapped[UUID] = mapped_column(primary_key=True, default=uuid4)
+    name: Mapped[str]
+    title: Mapped[str] = mapped_column(Text)
+    type: Mapped[Optional[AnnotationType]] = mapped_column(default=AnnotationType.text)
```
Comment (frascuchon) [inline src/argilla/server/models.py]: Why this attribute is optional? As far as I understand, optional attributes can be nullable, which differs from the table definition in this case

File: src/argilla/server/models.py [hunk 2]
```diff
@@ -34,26 +34,54 @@ def default_inserted_at(context):
...
+    required: Mapped[Optional[bool]] = mapped_column(default=False)
+    settings: Mapped[Optional[dict]] = mapped_column(JSON, default={})
```
Comment (frascuchon) [inline src/argilla/server/models.py]: Same for these fields
```suggestion
    required: Mapped[bool] = mapped_column(default=False)
    settings: Mapped[dict] = mapped_column(JSON, default={})
```

File: src/argilla/server/models.py (hunk 1 above)
Comment (jfcalvo) [reply src/argilla/server/models.py]: We have a default `AnnotationType.text`, that's why it's optional. If you don't specify one then we are using the `text` annotation type.

File: src/argilla/server/models.py (hunk 2 above)
Comment (jfcalvo) [reply src/argilla/server/models.py]: The same applies here.
* `required` has a default value of `False` so it's optional. If we don't specify a value for `required` the annotation then is not required. Of course we can do the opposite, if a value for `required` is not specified by default the annotation is required (using `True` as default).
* I was thinking that maybe we don't have a `settings` value for all kind of annotations so I'm using as default an empty JSON object here (or Python empty dict) that's why is optional too.

File: src/argilla/server/models.py (hunk 1 above)
Comment (jfcalvo) [reply src/argilla/server/models.py]: I have been rechecking and with a `default` value the `Optional` is useless. We can remove it.

File: src/argilla/server/models.py [hunk 3]
```diff
@@ -34,26 +34,54 @@ def default_inserted_at(context):
     return context.get_current_parameters()["inserted_at"]
 
 
+class AnnotationType(str, Enum):
+    text = "text"
+    single_label = "single_label"
```
Comment (frascuchon) [inline src/argilla/server/models.py]: So, after the slack conversation, we can change this to:
```suggestion
    free_text = "free_text"
    rating = "rating"
```

Comment (jfcalvo) [reply src/argilla/server/models.py]: I have added `rating` but didn't change `text` to `free_text`. Not sure if `free_text` it's better than simply `text`.

Comment (dvsrepo) [reply src/argilla/server/models.py]: I agree that simply `text` is better, the free doesn't add anything

Comment (frascuchon) [reply src/argilla/server/models.py]: Sorry, I reviewed this quickly, and I was thinking about single and multi-label when I did the changes. `text` fits well for me too.

**Review by frascuchon — APPROVED** [review]

**Pull request merged**